assets/src/components/detours/detourRouteSelectionPanel.tsx

```diff
@@ -84,14 +87,22 @@ export const DetourRouteSelectionPanel = ({
           <section className="pb-3">
             <h2 className="c-diversion-panel__h2">Choose direction</h2>
             {selectedRouteInfo.selectedRoute ? (
-              <RoutePropertiesCard
-                routePatterns={selectedRouteInfo.routePatterns}
-                selectedRoutePatternId={selectedRoutePatternFromInfo(
-                  selectedRouteInfo
-                )}
-                selectRoutePattern={onSelectRoutePattern}
-                defaultOpened="variants"
-              />
+              <div className="position-relative">
+                <RoutePropertiesCard
+                  routePatterns={selectedRouteInfo.routePatterns}
+                  selectedRoutePatternId={selectedRoutePatternFromInfo(
+                    selectedRouteInfo
+                  )}
+                  selectRoutePattern={onSelectRoutePattern}
+                  defaultOpened="variants"
+                />
+                <div
+                  hidden={!isLoadingRoutePatterns}
+                  className="position-absolute inset-0 bg-light opacity-75 d-flex justify-content-center align-items-center"
+                >
+                  <Spinner />
```

Comment:
Not exactly a nit, but a UX nit. This works GREAT if we're transitioning from a state with route patterns to another state with route patterns,

https://github.com/mbta/skate/assets/912020/107a02b5-5383-4f1f-8e95-ed35f22bbc73

...but is a much less smooth UX when you're starting with no route patterns

https://github.com/mbta/skate/assets/912020/958155c5-8d5f-4d36-8a0d-822b05c2d905

This can be a follow-up ticket, but I just wanted to flag it.